From 7e066db918aa9b7d8869871cb287ffe62232b71d Mon Sep 17 00:00:00 2001
From: John Fredriksson <john.fredriksson@intel.com>
Date: Thu, 9 Apr 2015 12:41:58 +0200
Subject: [PATCH] atomisp_gmin_platform: Remove legacy code for Whiskey Cove

Camera is now using VRF for PMIC stuff.

This is part of original patch:
https://android.intel.com/#/c/348536/ by Yunliang Ding

Split up in two for quilt reasons.

Note that this patch must be used together with quilt patch
"VRF: add camera consumer" to get camera working with
Whiskey Cove PMIC.

Change-Id: I5512a37453e3d83bdd14e7477592469e027761e1
Signed-off-by: John Fredriksson <john.fredriksson@intel.com>
---
 .../x86/platform/intel-mid/atomisp_gmin_platform.c | 28 +++-------------------
 1 file changed, 3 insertions(+), 25 deletions(-)

diff --git a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
index 3942894..95a4f16 100644
--- a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
+++ b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
@@ -28,7 +28,6 @@ EXPORT_SYMBOL(spid);
 #define DEVNAME_PMIC_AXP "INT33F4:00"
 #define DEVNAME_PMIC_TI  "INT33F5:00"
 #define DEVNAME_PMIC_CRYSTALCOVE "INT33FD:00"
-#define DEVNAME_PMIC_WHISKEYCOVE "INT34D3:00:6e"
 
 /* Should be defined in vlv2_plat_clock API, isn't: */
 #define VLV2_CLK_PLL_19P2MHZ 1
@@ -98,7 +97,7 @@ struct gmin_subdev {
 static struct gmin_subdev gmin_subdevs[MAX_SUBDEVS];
 
 static enum { PMIC_UNSET = 0, PMIC_REGULATOR, PMIC_AXP, PMIC_TI ,
-	PMIC_CRYSTALCOVE, PMIC_WHISKEYCOVE} pmic_id;
+	PMIC_CRYSTALCOVE } pmic_id;
 
 /* The atomisp uses type==0 for the end-of-list marker, so leave space. */
 static struct intel_v4l2_subdev_table pdata_subdevs[MAX_SUBDEVS+1];
@@ -337,8 +336,6 @@ static struct gmin_subdev *gmin_subdev_add(struct v4l2_subdev *subdev)
 			pmic_id = PMIC_TI;
 		else if (i2c_dev_exists(DEVNAME_PMIC_CRYSTALCOVE))
 			pmic_id = PMIC_CRYSTALCOVE;
-		else if (i2c_dev_exists(DEVNAME_PMIC_WHISKEYCOVE))
-			pmic_id = PMIC_WHISKEYCOVE;
 		else
 			pmic_id = PMIC_REGULATOR;
 	}
@@ -396,8 +393,8 @@ static struct gmin_subdev *gmin_subdev_add(struct v4l2_subdev *subdev)
 	}
 
 	if (pmic_id == PMIC_REGULATOR) {
-		gmin_subdevs[i].v1p8_reg = regulator_get(dev, "v1p8sx");
-		gmin_subdevs[i].v2p8_reg = regulator_get(dev, "v2p85sx");
+		gmin_subdevs[i].v1p8_reg = regulator_get(dev, "V1P8SX");
+		gmin_subdevs[i].v2p8_reg = regulator_get(dev, "V2P8SX");
 
 		/* Note: ideally we would initialize v[12]p8_on to the
 		 * output of regulator_is_enabled(), but sadly that
@@ -564,15 +561,6 @@ int gmin_v1p8_ctrl(struct v4l2_subdev *subdev, int on)
 								CRYSTAL_OFF);
 	}
 
-	if (pmic_id == PMIC_WHISKEYCOVE) {
-		int val = intel_soc_pmic_readb(WCOVE_V1P8SX_CTRL);
-		if (on)
-			return intel_soc_pmic_writeb(WCOVE_V1P8SX_CTRL,
-				(val & ~WCOVE_CTRL_MASK) | WCOVE_CTRL_ENABLE);
-		else
-			return intel_soc_pmic_writeb(WCOVE_V1P8SX_CTRL,
-				(val & ~WCOVE_CTRL_MASK) | WCOVE_CTRL_DISABLE);
-	}
 	return -EINVAL;
 }
 
@@ -629,16 +617,6 @@ int gmin_v2p8_ctrl(struct v4l2_subdev *subdev, int on)
 			return intel_soc_pmic_writeb(CRYSTAL_2P8V_REG, CRYSTAL_OFF);
 	}
 
-	if (pmic_id == PMIC_WHISKEYCOVE) {
-		int val = intel_soc_pmic_readb(WCOVE_V2P8SX_CTRL);
-		if (on)
-			return intel_soc_pmic_writeb(WCOVE_V2P8SX_CTRL,
-				(val & ~WCOVE_CTRL_MASK) | WCOVE_CTRL_ENABLE);
-		else
-			return intel_soc_pmic_writeb(WCOVE_V2P8SX_CTRL,
-				(val & ~WCOVE_CTRL_MASK) | WCOVE_CTRL_DISABLE);
-	}
-
 	return -EINVAL;
 }
 
-- 
1.9.1

